nixos/modules/services/networking/wireguard.nix

```diff
@@ -320,7 +320,14 @@ let
                 # cannot be used with systemd timers (see `man systemd.timer`),
                 # which is why `simple` with a loop is the best choice here.
                 # It also makes starting and stopping easiest.
+                #
+                # Retry even after "Name or service not known" dns failures:
+                Restart = "always";
```

Comment:
Could we only enable `Restart` for the "Name or service not known" error? It is possible if wireguard has a dedicated exit code for this error.

Reply:
Would we actually want that, though?
From my point of view, the primary intention of this PR is that wireguard should reconnect after it fails, pretty much no matter what.
If there are other failure cases than the dns failure that we are aware of here, we'd still want it to retry, right? (an unnesessary retry seems harmless, an avoidable permanent disconnect can be bad and require manual intervention)

Reply:
done